Source/objects.cpp

```diff
@@ -451,16 +451,14 @@ void AddCandles()
 void AddBookLever(Rectangle affectedArea, _speech_id msg)
 {
 	int cnt = 0;
-	int xp;
-	int yp;
+	Point position;
 	bool exit = false;
 	while (!exit) {
 		exit = true;
-		xp = GenerateRnd(80) + 16;
-		yp = GenerateRnd(80) + 16;
+		position = Point(GenerateRnd(80), GenerateRnd(80)) + Displacement { 16, 16 };
 		for (int n = -2; n <= 2; n++) {
 			for (int m = -2; m <= 2; m++) {
-				if (!RndLocOk(xp + m, yp + n))
+				if (!RndLocOk(position.x + m, position.y + n))
 					exit = false;
 			}
 		}

```

Comment:
Can use the point in rectangle range iterator for this :D
```cpp
for (Point nearPosition : PointInRectangleRange({ position, 2})) {
    if (!RndLocOk(nearPosition))
        exit = false;
}
```

Reply:
Going to do this as a follow up as it turns out to result in a lot of other changes that are better done as a separate cleanup.